novafacing/selene
# feat/cibuild

Lots of changes and updates! Incorporated @cxong's work on the GitHub workflow
and translated/updated their build script with MoonScript. Big thank you for the
original workflow.

This adds a workflow that will produce a working build of the game for each
architecture on pushes to `main` (use a dev branch!) as well as run unit tests.

- Create love-build.yml
- Update love-build.yml
- Update love-build.yml
- Add moonscript.init to package path for fused files
- Compile to lua
- Update love-build.yml
- Update love-build.yml
- Update conf.lua
- Created selenebuild build tool
- Add dist functionality
- Add git safety check to selenebuild
- Fixup conf file during distrib
- Reorganize to allow dirty check with lua artifact output
- Fix dist remove regex
- Update rocks in workflow and add test argument to selenebuild
- Add test runner to selenebuild and add error reporting
- Fix build failure reporting
- Add test step to love build workflow
- Fix typo
- Remove package path adjustments when building distribution
- Fix check logic
- Fix check logic
- Fix check logic
- Fix check logic
- Fixed typo
- Add line filtering and make it more generic
- Add debug
- Add blocklist line removal to selenebuild
- Add rockspec
- Switch to rockspec for installing dependencies for simplicity
- Update docs
